### DISCUSSION TITLE
How can one setup Rust lang completions using znap

### DISCUSSION THREAD

**Discussion by Pilgrim1379** [marlonrichert/zsh-snap#35] [Q&A] [answered]
So I'm slowly trying to move as much of zsh control over to znap and when I try to generate completions for rust with the code below I get the error `_arguments:comparguments:325: can only be called from completion function`

```
# rust and cargo completions
znap eval rust-completion 'rustup completions zsh'
znap eval cargo-completion 'rustup completions zsh cargo'
```
I'm wondering how you'll go about it.

**Comment by marlonrichert** [comment, marked as answer]
I created an issue to make this easier: #43

**Reply by marlonrichert** [reply]
@thepreacher It's implemented now on the `main` branch. Please try it out: https://github.com/marlonrichert/zsh-snap/commit/afb82095389cd9d862a813e1664ab02070b545bb#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5

**Reply by Pilgrim1379** [reply]
Just checked and it's working great. Thanks a bunch.
